engine/tests/phpunit/Elgg/Amd/ConfigTest.php

```diff
@@ -41,31 +52,39 @@ public function testCanRequireUnregisteredAmdModules() {
      * @expectedException \InvalidParameterException
      */
 	public function testThrowsOnBadShim() {
-		$amdConfig = new \Elgg\Amd\Config();
-		$amdConfig->addShim('bad_shim', array('invalid' => 'config'));
+		$this->amdConfig->addShim('bad_shim', array('invalid' => 'config'));
 
-		$configArray = $amdConfig->getConfig();
+		$configArray = $this->amdConfig->getConfig();
 
 		$this->assertEquals(array('jquery'), $configArray['deps']);
 	}
 
 	public function testCanAddModuleAsAmd() {
-		$amdConfig = new \Elgg\Amd\Config();
-		$amdConfig->addModule('jquery');
+		$this->amdConfig->addModule('jquery');
 
-		$configArray = $amdConfig->getConfig();
+		$configArray = $this->amdConfig->getConfig();
 
 		$this->assertEquals(array('jquery'), $configArray['deps']);
 	}
 
 	public function testCanAddModuleAsShim() {
-		$amdConfig = new \Elgg\Amd\Config();
-		$amdConfig->addModule('jquery.form', array('exports' => 'jquery.fn.ajaxform'));
+		$this->amdConfig->addModule('jquery.form', array('exports' => 'jquery.fn.ajaxform'));
 
-		$configArray = $amdConfig->getConfig();
+		$configArray = $this->amdConfig->getConfig();
 
 		$this->assertArrayHasKey('jquery.form', $configArray['shim']);
 		$this->assertEquals(array('exports' => 'jquery.fn.ajaxform'), $configArray['shim']['jquery.form']);
 	}
-}
 
+	public function testCheckAmdConfigHookTrigger() {
+		$test_input = ['test' => 'test_' . time()];
+		
+		$this->hooks->registerHandler('config', 'amd', function() {
+			return $test_input;
+		});
+		
+		$config = $this->amdConfig->getConfig();
+		
+		$this->assertEquals($test_input, $config);
```

Comment:
I can't remember if PHP considers "equivalent" arrays to be equal. You may need to use assertArrayEquals or something like that.